configure.ac

```diff
@@ -2621,12 +2621,22 @@ fi
 LIB_SOCKET_NSL
 AX_HARDEN_CC_COMPILER_FLAGS
 
-# link to ws2_32 if on mingw
+# if mingw then link to ws2_32 for sockets
 case $host_os in
-    *mingw32)
+    mingw*)
         LDFLAGS="$LDFLAGS -lws2_32" ;;
 esac
 
+# add wolfSSL defines for shared/static
+if test "$enable_shared" = "yes"
+then
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_SHARED"
+fi
+if test "$enable_static" = "yes"
+then
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_STATIC"
+fi
+
```

Comment:
Yes, in fact, autoconf is designed to build both shared and static by default for libraries.  That way if App X wants to link statically, and App Y wants to link dynamically, and App Z allows the user to pick how to link to the library everyone is happy.

options.h is designed for apps to have the same "settings" as the library.  I'm thinking this isn't a property of the library from the apps perspective since both types are allowed and supported.